dask/graphsurgery.py

```diff
@@ -0,0 +1,350 @@
+import uuid
```

Comment:
Happy to accept suggestions for a different module name, but I really think this should not be just added to dask.base, as it is all very advanced stuff. Also it would create a lot of circular dependencies.